utils/nctl/sh/scenarios/validators_disconnect.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+source "$NCTL"/sh/scenarios/common/itst.sh
+
+# Exit if any of the commands fail.
+set -e
+
+function main() {
+    log "------------------------------------------------------------"
+    log "Starting Scenario: Half of validators disconnecting"
+    log "------------------------------------------------------------"
+
+    # 0. Start the rest of the validator nodes
+    log_step "Starting nodes 6-10"
+    nctl-start node=6
+    nctl-start node=7
+    nctl-start node=8
+    nctl-start node=9
+    nctl-start node=10
+    # 1. Wait for the genesis era to complete
+    do_await_genesis_era_to_complete
+    # 2. Allow the chain to progress
+    do_await_era_change 1
+    # 3. Verify all nodes are in sync
+    parallel_check_network_sync 1 10
+    # 4. Stop half of the validators
+    log_step "Stopping nodes 6-10"
+    nctl-stop node=6
+    nctl-stop node=7
+    nctl-stop node=8
+    nctl-stop node=9
+    nctl-stop node=10
+    # 5. Wait for a period longer than the dead air interval
+    sleep_and_display_reactor_state '260.0'
+    # 6. Start 2 previously stopped validator node.
+    #    Now the network should have 7/10 active and start progressing.
+    log_step "Starting nodes 6, 7"
+    nctl-start node=6
+    nctl-start node=7
+    # 7. Check if the network is progressing
+    do_await_era_change_with_timeout 1 "500"
+    source "$NCTL"/sh/scenarios/common/health_checks.sh \
+            errors=0 \
+            equivocators=0 \
+            doppels=0 \
+            crashes=0 \
+            restarts=2 \
+            ejections=0
+
+    log "------------------------------------------------------------"
+    log "Scenario half of validators disconnecting complete"
+    log "------------------------------------------------------------"
+}
+
+function sleep_and_display_reactor_state() {
```

Comment:
Should we also assert that at the end of this function the reactor state is actually a `CatchUp`?

Reply:
Maybe a better way would be to add a function to check if the network has stalled (no blocks are produced anymore). Doesn't really matter what the reactor states are. I only had them printed because it helps with debugging.
I can add the check for network stall. WDYT?

Reply:
Added a step to check the network is actually stalled here https://github.com/casper-network/casper-node/pull/4120/commits/a980a885fe8d2f648071469ce4f53662b8b29326